src/Adapter/CustomerService/QueryHandler/GetCustomerThreadForViewingHandler.php

```diff
@@ -0,0 +1,386 @@
+                    [],
+                    'Admin.Catalog.Feature'
+                ),
+                'value' => CustomerThreadStatus::PENDING_1,
+            ];
+        } else {
+            $actions[CustomerThreadStatus::PENDING_1] = [
+                'label' => $this->translator->trans('Disable pending status', [], 'Admin.Catalog.Feature'),
+                'value' => CustomerThreadStatus::OPEN,
+            ];
+        }
+
+        if ($thread->status !== 'pending2') {
+            $actions[CustomerThreadStatus::PENDING_2] = [
+                'label' => $this->translator->trans(
+                    'Mark as "pending 2" (will be answered later)',
+                    [],
+                    'Admin.Catalog.Feature'
+                ),
+                'value' => CustomerThreadStatus::PENDING_2,
+            ];
+        } else {
+            $actions[CustomerThreadStatus::PENDING_2] = [
+                'label' => $this->translator->trans('Disable pending status', [], 'Admin.Catalog.Feature'),
+                'value' => CustomerThreadStatus::OPEN,
+            ];
+        }
+
+        return $actions;
+    }
+
+    /**
+     * @param CustomerThread $thread
+     *
+     * @return CustomerInformation
+     */
+    private function getCustomerInformation(CustomerThread $thread)
+    {
+        if (!$thread->id_customer) {
+            return new CustomerInformation(null, null, null, $thread->email, null, null, null);
+        }
+
+        $customer = new Customer($thread->id_customer);
+        $orders = Order::getCustomerOrders($customer->id);
+
+        $totalOk = 0;
+        $ordersOk = [];
+
+        if ($orders && count($orders)) {
+            foreach ($orders as $key => $order) {
+                if ($order['valid']) {
+                    $ordersOk[] = $order;
+                    $totalOk += $order['total_paid_real'] / $order['conversion_rate'];
+                }
+
+                $orders[$key]['date_add'] = Tools::displayDate($order['date_add']);
+                $orders[$key]['total_paid_real'] = Tools::displayPrice(
+                    $order['total_paid_real'],
+                    new Currency((int) $order['id_currency'])
+                );
+            }
+        }
+
+        return new CustomerInformation(
+            $customer->id,
+            $customer->firstname,
+            $customer->lastname,
+            $thread->email,
+            count($ordersOk),
+            $totalOk ? Tools::displayPrice($totalOk, $this->context->currency) : $totalOk,
+            (new DateTime($customer->date_add))->format($this->context->language->date_format_lite)
+        );
+    }
+
+    /**
+     * @param CustomerThread $thread
+     *
+     * @return string|null
+     */
+    private function getContactName(CustomerThread $thread)
+    {
+        $contacts = Contact::getContacts($this->context->language->id);
+
+        $contact = null;
+
+        foreach ($contacts as $c) {
```

Comment:
```suggestion
        foreach ($contacts as $contact) {
```
No single-letter variable names, please 😉